challenges/cloud/sre.md

```diff
@@ -0,0 +1,376 @@
+* Ability to execute integration tests against the local Kubernetes cluster
+* GitHub Actions for every commit that verifies code quality, ensures the software builds, and all tests pass
+* Produce production ready releases (binaries and docker image)
+
+### Deployment
+
+* Create a helm chart for the service
+* Deploy releases of the API Server to a Kubernetes cluster (and possible dependencies)
+* Upgrade releases of the API Server to a Kubernetes cluster (and possible dependencies)
+* Include a production Deployment of this service, including but not limited to: Deployment, ServiceAccount, Service
+
+
+## Level 5
+
+### Design
+
+In addition to including details about the requirements below, the design
+document should include a section describing how this solution could be deployed
+to an AWS EKS cluster. Please include security considerations as well.
+
+### Server
+
+* HTTP API to retrieve the replica count of the Kubernetes Deployment
+* HTTP API to set the replica count of a the Kubernetes Deployment
+* HTTP API to get the list of available Deployments in the Kubernetes cluster
+* HTTP health check verifying Kubernetes connectivity
+* Extend the API to support a difference between current and desired state
+* Extend the Server to support the ability to show the differences between current and desired state
+* Secure connections with mTLS
+* Replace the HTTP API with gRPC
+* Extend the Server to support reconciling cluster state (i.e. an external actor changed the replica count manually)
+
+### Automation
+
+* Ability to build the server in a Docker container
+* Ability to run the server
+* Ability to execute integration tests against the local Kubernetes cluster
+* GitHub Actions for every commit that verifies code quality, ensures the software builds, and all tests pass
+* Produce production ready releases (binaries and docker image)
+* Deploy and manage a local Kubernetes Cluster
+* Deploy and upgrade to the local Kubernetes Cluster with no service interruption
+* GitHub Action driven workflows for Deployment
+
+### Deployment
+
+* Create a helm chart for the service
+* Deploy releases of the API Server to a Kubernetes cluster (and possible dependencies)
+* Upgrade releases of the API Server to a Kubernetes cluster (and possible dependencies)
+* Include a production Deployment of this service, including but not limited to: Deployment, Role, RoleBinding, ServiceAccount, Service
+
+# Guidance
+
+## Interview process
+
+The interview team will join the Slack channel. The team consists of the
+engineers who will be working with you. Ask them about the engineering culture,
+work and life balance, or anything else that you would like to learn about
+Teleport.
+
+Before writing the actual code, create a brief design document in markdown in Github
+and share with the team.
+
+This document should cover at least: design approach, trade-offs, scope,
+proposed API, resiliency, scalability and deployment considerations.
```

Comment:
This is somewhat repetitive of the original section, and makes the reader look in multiple places, For example:

- we talk about starting with a design document on line 58 and then introduce the design doc again here
- we mention splitting the code into 2-3 PRs on line 31 but then we talk about splitting into PRs again on line 239

Reply:
I don't disagree that in parts this section doesn't introduce new information, but as long as we are consistent I don't think it's such an issue to reinforce and condense our expectations under the Guidance section, particularly if it's just a single line. 

With that said, I understand the value of brevity, so I've pared down this to focus on the process and less on expectations, PTAL